pulser-core/pulser/sequence/sequence.py

```diff
@@ -2172,12 +2196,42 @@ def _validate_and_adjust_pulse(
         self, pulse: Pulse, channel: str, phase_ref: Optional[float] = None
     ) -> Pulse:
         channel_obj: Channel
+        detuning_map: DetuningMap | None = None
         if channel in self._schedule:
             channel_obj = self._schedule[channel].channel_obj
+            if isinstance(channel_obj, DMM):
+                detuning_map = cast(
+                    _DMMSchedule, self._schedule[channel]
+                ).detuning_map
         else:
             # Sequence is parametrized and channel is a dmm_name
-            channel_obj = self.device.dmm_channels[_dmm_id_from_name(channel)]
-        channel_obj.validate_pulse(pulse)
+            dmm_id = _dmm_id_from_name(channel)
+            channel_obj = self.device.dmm_channels[dmm_id]
+            dmm_idx = -1
+            for call in self._calls[1:] + self._to_build_calls:
+                if (
+                    call.name == "config_detuning_map"
+                    or call.name == "config_slm_mask"
+                ):
+                    # Check whether dmm_name matches with channel
+                    (
+                        current_dmm,
+                        current_det_map,
+                    ) = self._get_dmm_id_detuning_map(call)
+                    if current_dmm == dmm_id:
+                        dmm_idx += 1
+                        current_dmm_name = (
+                            current_dmm
+                            if dmm_idx == 0
+                            else current_dmm + f"_{dmm_idx}"
```

Comment:
As we've done before, it would be better if we don't spread these arbitrary conventions on naming throughout the code but rather keep them in the same place. If we ever have to change them, we're more likely to miss something if they are scattered around